Paper: Imprecise uncertainty management with uncertain number to facilitate trustful computations

leslieDLcy opened the pull request:
Submission of the draft paper: Imprecise uncertainty management with uncertain number to facilitate trustful computations

Hi, I am Sanhita Joshi, @sanhitamj  . I will serve as the editor for this submission. Reach out to me if any assistance needed.

leslieDLcy committed:
initial_draft

leslieDLcy committed:
Added DOIs where possible

ameyxd commented:
Inviting reviewers: @Tolulade-A and @ktzhao

Tolulade-A commented:
Hello everyone, @leslieDLcy, I'm Tolulade, a software data engineer. I'm one of the reviewers for this paper.

leslieDLcy commented:
> Hello everyone, @leslieDLcy, I'm Tolulade, a software data engineer. I'm one of the reviewers for this paper.

Hi @Tolulade-A  and @ktzhao, I'm a research fellow based at Uni. of Liverpool. Thank you for taking the time to review our paper. We appreciate your feedback and look forward to your insights. cheers.

leslieDLcy committed:
added more DOIs

Tolulade-A commented on papers/leslie_yu_chen/main.md:
```diff
@@ -0,0 +1,326 @@
+% or prediction
+It is vital to *know what you do not know* in terms of trustful modelling and predictions, suggesting that *all models are wrong but some are useful* on the condition of knowing their assumptions and applicability, hence building the credibility of the computational results. It is often a challenge, in modelling complex physical phenomena, to construct mathematical models in a quantitative manner ...
+
+The increasing awareness of the differentiation of aleatory and epistemic uncertainty arises the need for more expressive mathematical frameworks to reason with various forms of uncertainty [@ferson2004arithmetic; @ferson1996different]. Imprecise uncertainty frameworks [@destercke2008unifying], such as evidence theory, random set, possibility distributions, credal set, capacities are proposed to  ...
```
The manuscript uses advanced domain-specific terms like _("aleatory," "epistemic," "p-box," "Dempster-Shafer structures")_ often. While some are defined, others are not. Please ensure all acronyms and technical terms are spelt out and briefly explained upon first mention.

leslieDLcy replied:
Thank you for your careful reading of the manuscript and your helpful comment. 

We have reviewed the text and confirmed that all acronyms have been defined upon first appearance, and also, in the `abbreviations` section of `myst.yml` file.

For better readability, we have also added wikipedia links for terms such as "p-box" and "Dempster-Shafer stuctures" upon first appearance.  Besides, we put relevant citations next to many of these terms for the audience interested in finding additional details. For example, upon mentioning "aleatory" and "epistemic" uncertainty, we have put relevant citations at the end of the sentence.

Meanwhile, short explanations were provided in the text for "p-box," "Dempster-Shafer structures" upon their first appearance, as shown in `main.md` Line 33, as quoted below:

> "These constructs are closely related to each other: P-boxes can be considered as interval bounds on cumulative distributions and DSS can be deemed as a discrete distribution with interval quantiles."

While we strive for clarity, some technical terms are standard in the field and widely understood by most audience. Providing detailed explanations for every term is beyond the scope of this paper and would risk interrupting the flow of the manuscript.

Tolulade-A commented on papers/leslie_yu_chen/main.md:
```diff
@@ -0,0 +1,326 @@
+Given the available information, there exists two untenable extreme (exclusive) endpoints which prescribe an interval of trustfulness.
+The lack of uncertainty quantification presented in many deterministic numerical simulations constitutes the lower bound, whereas the overconfidence of the computation, through introduction of unwarranted assumptions about uncertainties not faithful to the state of knowledge, as the upper bound.
+
+We aim at a faithful management of uncertainty throughout the computational pipeline using `uncertain number`, a unified construct for uncertainty characterisation at different imprecision.
```
Consistent narrative: Though minimal, in a few places, the narrative switches between first-person plural ("we aim," "this paper presents"). You could choose a consistent style.

leslieDLcy replied:
Thanks for the comment. In line with common academic writing practices, we use first-person plural ("we") to describe the authors' actions and views, and "this paper" for describing the content of the manuscript. We have reviewed the text and believe the current usage appropriately balances readability and formality. 

However, we are happy to make further adjustments if the editor may prefer a different style @sanhitamj .

sanhitamj replied:
Currently, we don't have a style guide ready for this - but the reviewer and the author bring a good point here. I will discuss this with the Proceedings and get back about it.

sanhitamj replied:
@leslieDLcy agreed with your explanation here.

Tolulade-A commented on papers/leslie_yu_chen/main.md:
```diff
@@ -0,0 +1,326 @@
+:width: 600px
+A bivariate p-box with marginals $X \sim \mathcal{B}([4, 8], 3)$ and $Y \sim \mathcal{N}([0,2], 2)$, and Gaussian copula parameterised by $\rho_{XY} = -0.8$ [@gray2021probabilityboundsanalysis]. The upper of the plot displays the joint CDF with upper and lower surface and the bottom shows the sampled deviates which are bivariate intervals
+:::
+
```
Phrases like "imprecise uncertainty frameworks," "probability bounding approach," and "Fr'{e}chet bound" could confuse readers from outside the immediate field. Please add clarifying sentences or parentheticals, or move definitions forward in the text.

leslieDLcy replied:
Thank you for your careful reading of the manuscript and your helpful comment.

For better clarity and readability and without compromising the flow of the manuscript, we have added relevant citations to these phrases in their first appearance. 

See for example:
- "imprecise uncertainty frameworks," --> `main.md` Line 16, as quoted below:

> The increasing awareness of the differentiation of aleatory and epistemic uncertainty arises the need for more expressive mathematical frameworks to reason with various forms of uncertainty [@ferson2004arithmetic; @ferson1996different]. Imprecise uncertainty frameworks [@destercke2008unifying], such as evidence theory, random set, possibility distributions, credal set, capacities are proposed to reflect the situations when information is scarce, vague, conflicting or imprecise whereby precise distribution are hard to be defined.

- "probability bounding approach," --> `main.md` Line 36, as quoted below:

> @fig:uc_constructs visually illustrates the notion of an `uncertain number`, which is underpinned by a probability bounding approach [@Williamson_1990; @ferson2003constructing] that allows for a faithful representation of the state of empirical information. For example, it can be characterised as a real number (a *degenerate* of an interval) when there is no uncertainty, a precise distribution (a *degenerate* of a p-box) when there is abundant data, and a set of distributions (e.g. a p-box) when there is partial information.

- "Fr'{e}chet bound" --> `main.md` Line 142, as quoted below:

> P-box arithmetic is built upon generalised probability convolutions, which through further extensions cover a wide spectrum of arithmetic operations: unary transformations, binary operations between p-boxes or Dempster-Shafer structures, and general functions composed of a series of base operations. It also covers a wide spectrum of dependency structures which could be fully known (specified copula $C$), partially known (lower bound copula $\underline{C}$), or even unknown ($\text{Fr\'{e}chet}$ bounds [@oberkampf2004dependence]



Meanwhile, the idea of "probability bounding approach" was illustrated in the Section **a probability bounding approach on information constraints**.


- [### a probability bounding approach on information constraints](https://github.com/leslieDLcy/scipy_proceedings/blob/3bf2c173c922a6e0400dfe880c16b468c8f96f3e/papers/leslie_yu_chen/main.md#L45)

Tolulade-A commented on papers/leslie_yu_chen/main.md:
```diff
@@ -0,0 +1,326 @@
+### measurement imprecision
+
+Empirical data rarely come in perfect forms, especially for *in situ* measurements.
+Practical computations frequently deal with poor measurements with different imprecision, possibly arised in recording, transmission, communication or manipulation, etc [@ferson2007experimental].
```
I noticed the use of "arised" instead of "arising", do review.

leslieDLcy replied:
Thanks for spotting the grammar error. "arised" has been corrected as "arising" as suggested.

Tolulade-A commented on papers/leslie_yu_chen/main.md:
```diff
@@ -0,0 +1,326 @@
+:::
+
+
+### aggregation of uncertianty
```
Do you mean "uncertainty" instead of "uncertianty"  ? Please review

leslieDLcy replied:
Thanks for spotting the typo. "uncertianty" has been corrected as "uncertainty" as suggested.

Tolulade-A commented on papers/leslie_yu_chen/main.md:
```diff
@@ -0,0 +1,326 @@
+
+
+## Non-deterministic computations and uncertainty propagation
+
```
The mathematical exposition is generally okay but may leave non-specialists behind, especially in the probability bounds and propagation sections. You may consider a brief, intuitive summary for each major mathematical construct before introducing notation-heavy details.

leslieDLcy replied:
Thanks for the comment. Improving clarity is the authors's goal, so we have a brief summary and also added additional wikipedia links to further illustrate the key mathematical constructs. These additions appear when we first introduce the concept of uncertain number, which occurs before the referenced propagation section.

1. We indeed already have a brief intuitive summary for two major mathematical constructs, namely "interval", "p-box" and "Dempster-Shafer structure (DSS)" when we first introduce "uncertain number". I don't believe "probability distribution" will need additional explanations.

See `main.md` Line 33, as quoted below:

> These constructs are closely related to each other: an interval indicates a value imprecisely known within a range where no assumptions of likelihood about the enclosed values are made; P-boxes can be considered as interval bounds on cumulative distributions and a DSS can be deemed as a discrete distribution with interval quantiles.


2. To further prepare the audience, when we first introduce the concept of uncertain number, we have also added corresponding wikipedia links which explain in details about the composing mathematical constructs, such as p-box. 

See `main.md` Line 32, as quoted below.

> `Uncertain number` stands for a generalised representation that unifies several uncertainty constructs including intervals, probability distributions, [probability boxes (p-boxes)](https://en.wikipedia.org/wiki/Probability_box) and [Dempster-Shafer structures (DSS)](https://en.wikipedia.org/wiki/Dempster–Shafer_theory), plus real numbers.

Tolulade-A commented on papers/leslie_yu_chen/main.md:
```diff
@@ -0,0 +1,326 @@
+
+The bounding approach presents as a natural reflection of the state of epistemic uncertainty, which tightens the bounds with extra empirical information. 
+As shown in @fig:characterisation_constraints, the level of information specifies constraints to accordingly construct the uncertain number, which starts as little as a coarse estimated range, to moment information, further to the knowledge of shape such that uncertainties are pinched to a precise distribution. These bounds are best-possible[^footnote-3] in the sense that they could not be any ti ...
+
```
`PyUncertainNumber` library is referenced a lot (>8), but no code snippets or concrete usage examples are provided. You could add a code block (Python) demonstrating a typical workflow with this library (e.g., constructing an uncertain number, propagating uncertainty, interpreting results).

Tolulade-A replied:
Also, you could consider a short section or figure summarising the reproducibility workflow.

leslieDLcy replied:
To facilitate clearer illustration and reproducibility of the workflow, essential code snippets are added in the main text, see the code snippets at section 2.1, 2.2, 2.3, 2.4, and section 3.2. Of which, the code snippet in section 3.2 briefly summarises a typical workflow of uncertainty characterisation and propagation.

Meanwhile, we have also added a comprehensive code tutorial, in the form of a Jupyter notebook, in the Supporting Documents section which demonstrates the details of the typical workflow of uncertainty characterisation, aggregation and propagation

The existence of the code snippets and the tutorial notebook has been made clear to the audience at the end of the Introduction section. 

See `main.md` Line 22, as quoted below:

> To facilitate clearer illustration and reproducibility of the workflow, essential code snippets are included in the main text, while a comprehensive tutorial is available in the Supporting Documents section.

**We only have a concern about the page limits (8 pages).** Currently the manuscript is fairly long and it is hard to tell precisely how many pages we already have from the webpage view, but it appears to be quite close, if not already exceeding.

sanhitamj replied:
We can overlook the 8 pages limit - now that Proceedings are published in the web-first fashion using CurveNote.

Tolulade-A commented on papers/leslie_yu_chen/main.md:
```diff
@@ -0,0 +1,326 @@
+title: Imprecise uncertainty management with uncertain number to facilitate trustful computations
+abstract: |
+  Scientific computations of complex systems are surrounded by various forms of uncertainty,  requiring appropriate treatment to maximise the credibility of computations. Empirical information for characterisation is often scarce, vague, conflicting and imprecise, requiring expressive uncertainty structures for trustful representation, aggregation and propagation. Current practices may present tw ...
+---
```
A short comparative paragraph or table somewhere in the paper could be of help. For example, how does the approach intended in this paper differ from/extend existing uncertainty quantification tools in Python?

leslieDLcy replied:
Thanks for the helpful comment.

As suggested, we have added a short comparative paragraph to summarise the advantage of our library which focuses on imprecise uncertainty management in **Section 5 - Conclusion and Outlook**:

See `main.md` Line 387, as quoted below:

>The developed Python library, `pyuncertainnumber`, facilitates trustworthy management of uncertainty through faithful representation and rigorous propagation. Unlike other established Python-based general purpose uncertainty quantification (UQ) tools that primarily focus on probabilistic modelling -—- often requiring precise specification of distribution shape and dependency structures, which is challenging in practice in the face of partial information, `pyuncertainnumber`, however, relaxes such specification and is capable of rigorous treatment even with unknown knowledge states. It explicitly addresses both aleatory and epistemic uncertainty using appropriate methods without making unjustified assumptions, advancing uncertainty management to handle both variability and incertitude. Consequently, it enables a more robust framework for addressing real-world UQ challenges such as mixed uncertainty propagation and characterisation under partial knowledge.

Together with other parts of the section, the advantages of the library are clearly spelled out: 

See`main.md` Line 379 - 387, as quoted below:

> Current practices tend to go overboard with unjustified assumptions of Gaussianality and independence, mostly due to the computational simplicity and the lack of tools for a comprehensive uncertainty analysis. This leads to a need of a computational framework that effectively balances representational expressiveness and computational feasibility.

> This paper presents the framework of `uncertain number` which fills this gap and presents several advantages:
> (i) it is highly expressive enabling faithful characterisation of an uncertain quantity given various scenarios of partial knowledge where conventional probability theory struggles to cope; 
> (ii) it provides a closed environment  where various characterisation, aggregation and propagation operations can be consistently conducted based on the unified structure, `uncertain number`;
> (iii) it is underpinned by a probability bounding mechanism which intuitively showcases the notion of epistemic uncertainty --- wherein increased knowledge leads to progressively tighter bounds;
> (iv) the bounding mechanism enables an explicit differentiation of aleatory and epistemic uncertainties, allowing their respective contributions to be  accounted for during both characterisation and propagation.

>The developed Python library, `pyuncertainnumber`, facilitates trustworthy management of uncertainty through faithful representation and rigorous propagation. Unlike other established Python-based general purpose uncertainty quantification (UQ) tools that primarily focus on probabilistic modelling -—- often requiring precise specification of distribution shape and dependency structures, which is challenging in practice in the face of partial information, `pyuncertainnumber`, however, relaxes such specification and is capable of rigorous treatment even with unknown knowledge states. It explicitly addresses both aleatory and epistemic uncertainty using appropriate methods without making unjustified assumptions, advancing uncertainty management to handle both variability and incertitude. Consequently, it enables a more robust framework for addressing real-world UQ challenges such as mixed uncertainty propagation and characterisation under partial knowledge.


In further notes, **Section 2 - Expressive power of uncertain number** demonstrates several examples of imprecise information where established Python-based UQ tools struggle to cope with, and **Section 3 - Non-deterministic computations and uncertainty propagation** demonstrates how mixed type uncertainty can be rigorously propagated even without knowing interdependencies of input variables, which established Python-based UQ tools struggle to cope with.

Tolulade-A commented on papers/leslie_yu_chen/myst.yml:
```diff
@@ -0,0 +1,55 @@
+  # Ensure your title is the same as in your `main.md`
+  title: Imprecise uncertainty management with uncertain number to facilitate trustful computations
+  # subtitle:
+  description: This paper demonstrates the framework of uncertain number which allows for a closed computation ecosystem whereby trustful computations can be conducted in a rigorous manner. This paper presents an overview of the main capabilities of the library PyUncertainNumber.
```
The paper references [PyUncertainNumber](https://pyuncertainnumber.readthedocs.io/en/latest/index.html), but there is no explicit citation of its repository. Please formally cite the software and provide an explicit link to the source code.

leslieDLcy replied:
Thanks for the helpful comment. 

We have added a footnote to the first mention of the package in the main text, which contains markdown links to both the source code repository and also the documentation.

See Line 395 in `main.md`, as quoted below:

> 
> `pyuncertainnumber` is a research-grade open-source library, with its source code available on [GitHub](https://github.com/leslieDLcy/PyUncertainNumber) and [comprehesive documentation](https://pyuncertainnumber.readthedocs.io/en/latest/index.html) available online.
> 

We previously had a markdown link to the documentation of the package in the first mention in the main text, as it contains a lot of helpful information, such as explanatory pages to those advanced terms such as p-box, and also the link to the source code repository of course. Given the suggestion of the reviewer, we added both the links to source code and documentation.

Tolulade-A commented on papers/leslie_yu_chen/main.md:
```diff
@@ -0,0 +1,326 @@
+
+The developed Python library, `PyUncertainNumber`, facilitates trustful management of uncertainty through faithful representation and rigorous propagation. As probabilistic programming provides support for automatic inference, we aim for an imprecise uncertainty analysis framework, where variables are consistently represented by uncertain numbers in the face of both variability and incertitude, a ...
+Our next focus will be extending the framework of probabilistic programming, which has a heavy focus on Bayesian inference, into an imprecise realm where more comprehensive uncertainty structures can be integrated into the learning and inference of machine learning models.
+
```
You could add an example (code snippet) showing how to reproduce at least one of the main results or figures in the paper.

leslieDLcy replied:
Thanks for the helpful comment.

We have added essential code snippets in the main text for reproducing Fig. 2, Fig. 3, Fig. 4 and Fig. 5 right down below these figures. Trivial details in terms of styling aesthetics (e.g. line colour, fonts, etc) are omitted to keep the main text concise. 

However, such details can be found in the tutorial notebook in the Supporting Documents, for example see tutorial notebook: Section 3 for reproducing Fig.3 and Section 4.1 for reproducing Fig. 7.

leslieDLcy committed:
fixed DOI one more time and chaned three .pdf figures into .png figures for a stable building process

Tolulade-A commented on papers/leslie_yu_chen/main.md:
```diff
@@ -0,0 +1,326 @@
+:::{figure} imprecise_measurements_combo.pdf
+:label: fig:imprecise_measurements
+:width: 800px
+Characterisation of imprecise measurements. **(a)**: fitting an exponential distribution to an interval datum by maximum likelihood estimation; **(b)**: Parametric and nonparamettric characterisation of the whole imprecise data set which includes 15 data points i.i.d (independent and identically distributed) sampled from Exp(0.4) contaminated by a margin of error $\Delta=1.4$
```
Data sources for the examples and figures are not specified (e.g., `15 data points i.i.d sampled from Exp(0.4)`contaminated by a margin of error`. If synthetic, clarify; if from real data, provide a citation.

leslieDLcy replied:
Thanks for the comment. But in fact, the data source has been specified literally as quoted "15 data points i.i.d sampled from Exp(0.4)contaminated by a margin of error $\Delta=1.4$" where the mathematical notation `Exp(0.4)` stands for an [exponential distribution](https://en.wikipedia.org/wiki/Exponential_distribution) parameterised by 0.4. Therefore,  it is synthetic data.

For better clarification, we have revised the figure caption to explicitly explain the mathematical notation as well as the data generating process. Besides, we have revised the description to the figure in the main text to further explain.

See:

- revised figure caption, as seen in `main.md`, Line 116:

> Characterisation of imprecise measurements. **(a)** fitting an exponential distribution to an interval datum by maximum likelihood estimation; **(b)** Parametric and nonparamettric characterisation of the whole imprecise data set $\{ x_{i}\}$ which includes 15 data points i.i.d (independent and identically distributed) sampled from an exponential distribution (i.e. $\text{Exp}(0.4)$) contaminated by a margin of error $\Delta=1.4$. That is, $x_{i} = [\underline{x}_i, \overline{x}_i] = [m_{i} - \Delta, m_{i} + \Delta]$ where $m_i \sim \text{Exp}(0.4)$.

- revised description of the figure in the main text, as seen `main.md` Line 104:

> For example, as shown in @fig:imprecise_measurements [a], a set of maximum likelihood estimates $\lambda^{I} = [\underline{\lambda}, \overline{\lambda}]$ are yielded for a single datum where $\lambda$ indicates the parameter of exponential distribution. Collectively, the dataset leads to an interval bound of the fitted exponential distribution shown in the shaded area in @fig:imprecise_measurements [b], along with the true data generating mechanism (DGM) in red dotted curve and the empirical CDF in gray dotted curve. Further, as a nonparametric comparison, the Kolmogorov Smirnov confidence bands is also extended [@tretiak2023should], as shown in blue dashes below. These are both rigorous uncertain numbers that enclose the true yet unknown data generating distribution.

Tolulade-A commented on papers/leslie_yu_chen/main.md:
```diff
@@ -0,0 +1,326 @@
+Often there is little empirical information pertaining some parameters of a mathematical model. A faithful characterisation entails that all of the available statistical information should be utilised but without introducing any extra assumptions beyond what are empirically justified.
+
+The bounding approach presents as a natural reflection of the state of epistemic uncertainty, which tightens the bounds with extra empirical information. 
+As shown in @fig:characterisation_constraints, the level of information specifies constraints to accordingly construct the uncertain number, which starts as little as a coarse estimated range, to moment information, further to the knowledge of shape such that uncertainties are pinched to a precise distribution. These bounds are best-possible[^footnote-3] in the sense that they could not be any ti ...
```
Some acronyms like `CDF` here and `DSS, DGM, KS, DOE, PBA, UQ, CFD, API` in other places are listed in the YAML but not always spelt out **on first use in the script**. Please ensure every acronym is expanded at first mention.

leslieDLcy replied:
Thanks for the comment. All the acronym has been referenced in the `abbreviations` section of the `myst` file, enabling the spelt-out text to be shown when hovering on the acronym. I believe this is the intended writing style from the organisers.

Tolulade-A commented on papers/leslie_yu_chen/main.md:
```diff
@@ -0,0 +1,326 @@
+Intervals turn out to be natural constructs for representing *incertitude* in imprecise measurements, manifested either in a direct interval or a plus-or-minus form. When interval-valued measurements are present in a data set, a single probability distribution is inadequate to characterise the epistemic uncertainty. 
+Rather, the bounding strategy applies whereby classical inference methods are extended to both characterise the sampling uncertainty and also data imprecision. For example, as shown in @fig:imprecise_measurements [a], a set of maximum likelihood estimates $\lambda^{I} = [\underline{\lambda}, \overline{\lambda}]$ are yielded for a single datum. Collectively, the dataset leads to an interval bound  ...
+Further, as a nonparametric comparison, the Kolmogorov Smirnov confidence bands is also extended [@tretiak2023should], as shown below. These are both rigorous uncertain numbers that enclose the true yet unknown data generating distribution.
+
```
Some of the mathematical symbols are introduced without formal definitions; they could be defined clearly before use.

leslieDLcy replied:
Thanks for the comment. Explicit definition of $\lambda$ has been added for better clarity as suggested. 

The corresponding sentence has been revised as:

> For example, as shown in @fig:imprecise_measurements [a], a set of maximum likelihood estimates $\lambda^{I} = [\underline{\lambda}, \overline{\lambda}]$ are yielded for a single datum where $\lambda$ indicates the parameter of exponential distribution.

leslieDLcy commented:
The authors appreciate the comments from the reviewer @Tolulade-A , which we believe will make the manuscript more readable and clearer. We will start addressing the reviews shortly. cheers.

ktzhao commented on papers/leslie_yu_chen/mybib.bib:
```diff
@@ -0,0 +1,354 @@
+  month=may, 
+  pages={107405} 
+}
+
```
It is a pleasure and an honor to review your manuscript. My apologies for the delayed response due to other professional commitments.
 I noticed that the citation agrell2024nasa lists the publication year as 2024。However, the article was actually published in 2021. To avoid potential confusion for readers, would you consider updating the citation key to reflect the correct publication year?

leslieDLcy replied:
Pleasure is mine! The authors appreciate the feedback and pointing out the lapse. We will start addressing your comment shortly.

leslieDLcy replied:
Thank you for your careful reading of the manuscript and pointing out this error. 

The citation has been corrected as suggested, which is shown below

```bibtex
@article{agrell2021nasa, 
  title={The NASA langley challenge on optimization under uncertainty},
  author={Crespo, Luis G and Kenny, Sean P},
  volume={152}, 
  ISSN={0888-3270}, 
  url={http://dx.doi.org/10.1016/j.ymssp.2020.107405}, 
  DOI={10.1016/j.ymssp.2020.107405}, 
  journal={Mechanical Systems and Signal Processing}, 
  publisher={Elsevier BV}, 
  year={2021}, 
  month=may, 
  pages={107405} 
}

ameyxd commented:
@sanhitamj will serve as editor for this paper.

leslieDLcy committed:
Replaced \scriptsize{f} as just {f} as an effort to avert potential issues in rendering

leslieDLcy committed:
fixed typos and grammar errors

leslieDLcy committed:
some more revisions

leslieDLcy committed:
remove redundance of words

leslieDLcy committed:
revise the notes in the code snippet

leslieDLcy committed:
added acknowledgement

leslieDLcy committed:
added a tutorial notebook file for reproducible workflow

leslieDLcy committed:
added text regarding inline code snippets and Tutorial notebook in the Supporting Documents Section

leslieDLcy committed:
fixed the heading level lapse

leslieDLcy committed:
simplify inline code snippets due to page limits

leslieDLcy committed:
revise a repo in the tutorial notebook

sanhitamj commented on papers/leslie_yu_chen/main.md:
```diff
@@ -0,0 +1,399 @@
+It is vital to *know what you do not know* in terms of trustworthy modelling and predictions, suggesting that *all models are wrong but some are useful* on the condition of knowing their assumptions and applicability, hence building the credibility of the computational results. It is often a challenge, in modelling complex physical phenomena, to construct mathematical models in a quantitative man ...
+
+The increasing awareness of the differentiation of aleatory and epistemic uncertainty arises the need for more expressive mathematical frameworks to reason with various forms of uncertainty [@ferson2004arithmetic; @ferson1996different]. Imprecise uncertainty frameworks [@destercke2008unifying], such as evidence theory, random set, possibility distributions, credal set, capacities are proposed to  ...
+Given the available information, there exists two untenable extreme (exclusive) endpoints which prescribe an interval of trustworthyness.
```
Grammar: ... there exist two untenable extreme ....

leslieDLcy replied:
Thanks for spotting this grammar error. It has been revised.

sanhitamj commented on papers/leslie_yu_chen/main.md:
````diff
@@ -0,0 +1,399 @@
+
+```{code-block} python
+
+from pyuncertainnumber import pun
````
The comment is just to appreciate the `pun`.

sanhitamj replied:
@leslieDLcy I enjoyed reading the paper. I have shared the preview of the paper with a few Data Scientists at work, who enjoy word play and are readers. 

Thank you so much for thinking about this topic and the submission. Especially section 2.4

leslieDLcy replied:
Many thanks! Happy to hear you enjoyed it. Hope it can spark some interest among your colleagues too.  Stay in touch.

sanhitamj commented on papers/leslie_yu_chen/main.md:
```diff
@@ -0,0 +1,399 @@
+
+## Aggregation of uncertainty
+
+One of the controversial subject in uncertainty analysis is the aggregation of multiple (imperfect) sources of information, evidence, or expert elicitations. Information varies in quality and could be conflicting. Expert may have different degrees of subjectivity and different representations for the uncertain quantity, leading to epistemic uncertainty in the various educated estimates. 
```
Grammar: Expert**s** may have ...

leslieDLcy replied:
Thanks for spotting this grammar error. It has been revised.

leslieDLcy committed:
revised a typo

leslieDLcy committed:
revised some typos

leslieDLcy committed:
added acronym for NLP

leslieDLcy committed:
added comparison to extablished Python UQ packages

leslieDLcy committed:
added concise interval explanation

leslieDLcy committed:
revised a lapse

leslieDLcy committed:
added two important citations

leslieDLcy committed:
fixed a lapse in adding citations

leslieDLcy committed:
fix the author name error from Zenodo

leslieDLcy committed:
fixed an issue with a figure

leslieDLcy commented:
Dear Co-Chair @sanhitamj, I hope this message finds you well. I just wanted to kindly check in on how the reviewing and publication process is progressing. There appears no update for a while. 

In response to the [issue](https://github.com/scipy-conference/scipy_proceedings/issues/1082#issue-3121982861) mentioned earlier in which `\mathbb` breaks the typst PDF export, I have committed a simple fix which replaces `\mathbb` with `\mathbf`. I hope this may fix the issue though it seems not 🥹 .
> [ ] Support nested macros in `\mathbb` latex macro in typst export (currently the macro is treated as text and duplicated which breaks the PDF export) [Paper: Imprecise uncertainty management with uncertain number to facilitate trustful computations #1099](https://github.com/scipy-conference/scipy_proceedings/pull/1099)

With this proposed fix, the GitHub Action runs okay but still the PDF export has not shown in the Preview.

leslieDLcy committed:
Workaround for the small issue breaking PDF export

In response to the [issue](https://github.com/scipy-conference/scipy_proceedings/issues/1082#issue-3121982861) in which the use of `\mathbb` latex macro in typst export
is treated as text and duplicated which breaks the PDF export.

This commit provides a simple workaround by replacing "\mathbb" with "\mathbf".

leslieDLcy committed:
removed suspicious \mathbf that may have resulted the myst pbox export issue

leslieDLcy committed:
changed back \mathbb since it is said to be fixed
